### ISSUE TITLE
Tree option causes stack overflow on macOS with root privileges

### ISSUE THREAD

**Issue by TheAlgorythm** [dalance/procs#129]
```zsh
$  procs -V
procs 0.11.4
$  procs -t #works
$  sudo procs #works
$  sudo procs -t

thread 'main' has overflowed its stack
fatal runtime error: stack overflow
[1]    16768 abort      sudo procs -t
$  ulimit -s 65532
$  sudo procs -t

thread 'main' has overflowed its stack
fatal runtime error: stack overflow
[1]    16847 abort      sudo procs -t
```

**Comment by kidonng** [comment]
A stacktrace, hope it helps:
[procs_2021-05-02-232600_Pavilion.txt](https://github.com/dalance/procs/files/6412762/procs_2021-05-02-232600_Pavilion.txt)

**Comment by dalance** [comment]
Thanks. Could you provide pid and ppid list when the issue happens?

**Comment by kidonng** [comment]
Very interestingly, I can't reproduce the issue for now. Will try provide one if I can.

**Comment by TheAlgorythm** [comment]
On my system it is reproducible and here is the list of processes. [processes.log](https://github.com/dalance/procs/files/6427413/processes.log)

I extracted it with the following command and I hope it works as intended.
```zsh
$ sudo ps -A -o pid,ppid > processes.log
```

**Comment by dalance** [comment]
I couldn't reproduce it, but found a problematic code.
Could you try the latest master branch?

**Comment by TheAlgorythm** [comment]
Perfect 🤩 It works. I checked the version before your commit and there, it fails.

**Comment by dalance** [comment]
Thanks!
I released v0.11.5.
